.github/workflows/ci.yml

```
@@ -1,152 +1,84 @@
 name: ci
 on:
+  pull_request:
   push:
     branches:
     - master
-  pull_request:
-    types:
-    - opened
-    - reopened
-    - synchronize
 jobs:
-  linux-tests:
-    name: linux tests
-    runs-on: ubuntu-18.04
+  tests:
+    name: tests
+    strategy:
+      matrix:
+        # os: [ubuntu-18.04, macos-10.14, windows-2016]
+        os: [ubuntu-18.04, macos-10.14]
+        # channel: [stable, beta, nightly]
+        channel: [nightly]
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
     env:
-      DISPLAY: ":99"
       ATOM_GITHUB_BABEL_ENV: coverage
       MOCHA_TIMEOUT: 60000
       UNTIL_TIMEOUT: 30000
     steps:
     - uses: actions/checkout@v1
     - name: install Atom
-      shell: bash
+      uses: UziTech/action-setup-atom@v1
+      with:
+        channel: ${{ matrix.channel }}
+    - name: install windows build tools
       run: |
-        curl -s -L https://atom.io/download/deb?channel=nightly \
-          -H 'Accept: application/octet-stream' \
-          -o /tmp/atom-amd64.deb
-        /sbin/start-stop-daemon --start --quiet \
-          --pidfile /tmp/custom_xvfb_99.pid \
-          --make-pidfile --background \
-          --exec /usr/bin/Xvfb \
-          -- :99 -ac -screen 0 1280x1024x16
-        sudo apt-get update
-        sudo apt-get install -yyq libgconf-2-4 build-essential git libsecret-1-dev gvfs-bin
-        dpkg-deb -x /tmp/atom-amd64.deb /tmp/atom
+        choco install vcbuildtools
```

Visual Studio is already installed on the GitHub images: https://github.com/actions/virtual-environments

---

Windows tests weren't working without that install so maybe at the time this PR was created they weren't automatically installed? I'll wait to remove it until this PR actually gets some attention. By that time many other parts might need to be changed.

---

Windows-latest has the 2019 build tools. Maybe It is because you have to install 2015 or 2017 version. Choco installs 2015 with this command:
https://chocolatey.org/packages/vcbuildtools

---

I agree, if it works with that it would be better

---

```suggestion
        choco install visualstudio2017buildtools -y
```

---

This isn't really worth working on unless someone at @github or @atom is actually considering merging it.

---

This is the fastest way I found to install visual studio 2015:
```suggestion
      - name: Install Visual Studio 2015 on Windows
        if: ${{ contains(matrix.os, 'windows') }}
        run: |
          choco install visualcpp-build-tools --version=14.0.25420.1 --ignore-dependencies -y --params "'/IncludeRequired'"
          echo ::set-env name=VCTargetsPath::'C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\v140'
```

It is used in language-javascript: 
https://github.com/atom/language-javascript/blob/e5a7d95edfca40dcae78fd536fff04393f5fe3c0/.github/workflows/ci.yml#L34